src/App.tsx

```diff
@@ -1,109 +1,75 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { useSearchParams } from 'react-router-dom';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const items: string[] = getNumbers(1, 42).map(n => `Item ${n}`);
+
+enum ItemsPerPage {
+  three = 3,
+  five = 5,
+  ten = 10,
+  twenty = 20,
+}
 
 export const App: React.FC = () => {
+  const [, setSearchParams] = useSearchParams();
+  const [perPage, setPerPage] = useState(ItemsPerPage.five);
+  const [currentPage, onPageChange] = useState(1);
+
+  const itemsOnPage = items.slice(
+    (currentPage - 1) * perPage,
+    (currentPage - 1) * perPage + perPage,
+  );
+
+  useEffect(() => {
+    setSearchParams({
+      page: currentPage.toString(),
+      perPage: perPage.toString(),
+    });
+  }, [currentPage, perPage, setSearchParams]);
```

Comment:
It's good practice to synchronize the URL search parameters with the app state, but be aware that this can lead to performance issues if not done carefully. Since `setSearchParams` is an expensive operation, make sure to debounce or throttle this effect if you notice performance problems, especially when dealing with a large number of updates in a short time.